service/src/test/java/uk/gov/hmcts/reform/fpl/service/cmo/UploadCMOServiceTest.java

```diff
@@ -415,284 +415,61 @@ void shouldUpdateExistingCMOWithNewOrderAndChangeStatus() {
     @Test
     void shouldBuildAgreedEventWhenNewCMOIsAgreed() {
         List<Element<CaseManagementOrder>> unsealedOrders = List.of(
-            element(CaseManagementOrder.builder().status(DRAFT).build())
+            element(CaseManagementOrder.builder().status(SEND_TO_JUDGE).build())
         );
 
-        List<Element<HearingBooking>> hearings = new ArrayList<>(hearings());
-        HearingBooking updatedHearing = hearings.get(0).getValue().toBuilder()
-            .caseManagementOrderId(unsealedOrders.get(0).getId())
-            .build();
-        hearings.set(0, element(hearings.get(0).getId(), updatedHearing));
+        List<Element<HearingBooking>> hearingsBefore = new ArrayList<>(hearings());
 
         CaseData caseDataBefore = CaseData.builder()
-            .hearingDetails(hearings)
-            .build();
-
-        CaseData caseData = CaseData.builder()
-            .hearingDetails(hearings)
-            .draftUploadedCMOs(unsealedOrders)
+            .hearingDetails(hearingsBefore)
             .build();
 
-        UploadCMOEvent event = service.buildEventToPublish(caseData, caseDataBefore);
-
-        assertThat(event).isEqualToComparingFieldByField(new AgreedCMOUploaded(caseData, updatedHearing));
-    }
-
-    @Test
-    void shouldBuildDraftEventWhenNewCMOIsDraft() {
-        List<Element<CaseManagementOrder>> unsealedOrders = List.of(
-            element(CaseManagementOrder.builder().status(DRAFT).build())
-        );
+        List<Element<HearingBooking>> hearingsAfter = new ArrayList<>(hearingsBefore);
 
-        List<Element<HearingBooking>> hearings = new ArrayList<>(hearings());
-        HearingBooking updatedHearing = hearings.get(0).getValue().toBuilder()
+        HearingBooking updatedHearing = hearingsAfter.get(0).getValue().toBuilder()
```

Comment:
maybe it would be useful to add to Element
    public Element copy(UnaryOperator<T> transformer) {
        return Element.builder()
            .id(this.id)
            .value(transformer.apply(this.value)).
            build();
    }

Reply:
or can be in ElementUtils or as a private method here, test should focus on what is done and not how, this seems for me a bit low level and used twice

Reply:
I like the idea of the copy method, would it be worthwhile adding an overloaded version that just copies without transforming?